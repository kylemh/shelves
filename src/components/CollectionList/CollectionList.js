import React from 'react';
import PropTypes from 'prop-types';
import DraggableReleaseItem from '../ReleaseItem/DraggableReleaseItem';
import ReleaseItemPropTypes from '../ReleaseItem/ReleaseItemPropTypes';
import styles from './CollectionList.module.scss';

export default class CollectionList extends React.Component {
  static propTypes = {
    collection: PropTypes.arrayOf(
      PropTypes.shape({
        ...ReleaseItemPropTypes,
      })
    ).isRequired,
    innerRef: PropTypes.func,
    placeholder: PropTypes.string,
  };

  static defaultProps = {
    innerRef: undefined,
    placeholder: undefined,
  };

  render() {
    const { collection, placeholder, innerRef, ...rest } = this.props;

    return (
      <section className={styles.CollectionList} ref={innerRef} {...rest}>
        {collection.map((release, index) => (
          <DraggableReleaseItem
            index={index}
            key={release.id}
            {...release}
          />
        ))}
        {placeholder}
      </section>
    );
  }
}
