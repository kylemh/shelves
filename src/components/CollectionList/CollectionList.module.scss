@import '../../styles/variables.scss';

.CollectionList {
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  height: 12rem;
  width: 100%;
  padding: 1rem 0;
  background-color: $feint;
  overflow-x: auto;
  overflow-y: hidden;
}
