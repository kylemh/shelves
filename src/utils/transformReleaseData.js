/**
 * @description Pass a release to this function and get back the relevant info for this UI
 * @param {object} release
 * @param {number} release.id
 * @param {object} release.basic_information
 * @param {number} release.basic_information.year
 * @param {string} release.basic_information.title
 * @param {[{ name: string }]} release.basic_information.artists
 * @exports
 * @returns
 */
function transformReleaseData({ id, basic_information: basicInfo }) {
  const { artists, title, year } = basicInfo;

  const primaryArtistName = artists[0].name;

  return {
    id,
    primaryArtistName,
    title,
    year,
  };
}

export default transformReleaseData;
