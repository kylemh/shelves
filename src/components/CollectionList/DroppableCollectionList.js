import React from 'react';
import PropTypes from 'prop-types';
import { Droppable } from 'react-beautiful-dnd';
import CollectionList from './CollectionList';
import ReleaseItemPropTypes from '../ReleaseItem/ReleaseItemPropTypes';

export default class DroppableCollectionList extends React.Component {
  static propTypes = {
    collection: PropTypes.arrayOf(
      PropTypes.shape({
        ...ReleaseItemPropTypes,
      })
    ).isRequired,
    droppableId: PropTypes.string.isRequired,
  };

  render() {
    const { collection, droppableId } = this.props;

    return (
      <Droppable droppableId={droppableId} direction="horizontal">
        {provided => (
          <CollectionList
            collection={collection}
            innerRef={provided.innerRef}
            placeholder={provided.placeholder}
            {...provided.droppableProps}
          />
        )}
      </Droppable>
    );
  }
}
