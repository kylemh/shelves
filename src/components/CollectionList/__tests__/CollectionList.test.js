import React from 'react';
import { shallow } from 'enzyme';

import transformReleaseData from '../../../utils/transformReleaseData';
import mockedReleases from '../../../api/mocks/releases-0.json';
import CollectionList from '../CollectionList';

const testData = mockedReleases.releases.slice(0, 3).map(release => transformReleaseData(release));

describe('CollectionList', () => {
  it('should render', () => {
    const wrapper = shallow(<CollectionList collection={testData} />);

    expect(wrapper).toMatchSnapshot();
  });
});
