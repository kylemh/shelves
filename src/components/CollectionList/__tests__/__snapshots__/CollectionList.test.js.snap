// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionList should render 1`] = `
<section
  className="CollectionList"
>
  <DraggableReleaseItem
    id={76035}
    index={0}
    key="76035"
    primaryArtistName="Swayzak"
    title="Dirty Dancing"
    year={2002}
  />
  <DraggableReleaseItem
    id={198657}
    index={1}
    key="198657"
    primaryArtistName="Jazkamer"
    title="Sound Of Music"
    year={2002}
  />
  <DraggableReleaseItem
    id={619145}
    index={2}
    key="619145"
    primaryArtistName="Tiga"
    title="Sexor"
    year={2006}
  />
</section>
`;
